yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/internal/tasks/GraphPlanner.java

```diff
@@ -114,7 +116,20 @@ public ForkTask plan(Step root) {
                         target.inputs.addAll(node.inputs);
                         target.available.addAll(node.available);
                         nodes.remove(key);
-                        modified = true;
+
+                        //check if all non-END nodes only have one todo
+                        boolean found = false;
+                        for (Map.Entry<Step, Node> entry : nodes.entrySet()) {
+                            if (entry.getKey() instanceof OperatorStep) {
+                                if (!((OperatorStep) entry.getKey()).getCompute().getOperator().equals(PhysicalOperator.END)) {
+                                    if (!(entry.getValue().todo.size() == 1)) {
+                                        found = true;
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                        modified = true && found;
```

Comment:
This could be 'modified = found'

Reply:
Let me  check. Thank you! About the test failures which jdk did you use?

Reply:
I was using JDK 11, which I suspect is why I was seeing failures.

Reply:
Merge after changing 'modified = found'. Thanks